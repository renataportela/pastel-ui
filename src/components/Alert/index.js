export { default, ALERT_COLORS } from './Alert'