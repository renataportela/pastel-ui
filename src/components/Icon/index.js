export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';