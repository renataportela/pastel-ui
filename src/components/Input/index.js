export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
