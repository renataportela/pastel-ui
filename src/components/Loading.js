import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading