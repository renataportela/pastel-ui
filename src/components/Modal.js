import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;