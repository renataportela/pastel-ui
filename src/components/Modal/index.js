export { default } from './Modal'
export { default as ModalBase } from './ModalBase'