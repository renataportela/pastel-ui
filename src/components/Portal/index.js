export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'