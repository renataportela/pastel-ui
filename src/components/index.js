export { default as Alert } from './Alert'
export { default as AppBar } from './AppBar'
export { default as Avatar } from './Avatar'
export { default as BackTop } from './BackTop'
export { default as Badge } from './Badge'
export { default as Button, Close } from './Button'
export { default as Checkbox } from './Checkbox'
export { default as Chip } from './Chip'
export { default as Dialog } from './Dialog'
export { default as Drawer } from './Drawer'
export { default as Form } from './Form'
export { default as FormField } from './FormField'
export { Heading, Paragraph, Text } from './Typography'
export { default as Icon, Svg } from './Icon'
export { default as Menu } from './Menu'
export { default as Modal } from './Modal'
export { default as Overlay } from './Overlay'
export { default as PageLoading } from './PageLoading'
export { default as Pagination } from './Pagination'
export { default as Paper } from './Paper'
export { default as Portal } from './Portal'
export { default as Progress, CircularProgress } from './Progress'
export { default as Radio } from './Radio'
export { default as Tag } from './Tag'
export { default as TextField } from './TextField'
export { default as Theme, BaseColors, useTheme } from './Theme'
export { default as Toast, useToast, withToastProvider } from './Toast'
export { default as Ripple } from './Ripple'

export * from './Grid'