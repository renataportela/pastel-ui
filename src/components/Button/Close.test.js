import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, Close } from '~/components'

describe('Close component', () => {
  it('renders a close button with large size', () => {
    const tree = renderer
      .create(<Theme><Close size="lg" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})