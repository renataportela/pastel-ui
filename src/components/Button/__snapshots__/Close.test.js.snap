// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close component renders a close button with large size 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 wdmuf"
  color="primary"
  kind="ghost"
  size="lg"
>
  <svg
    className="Svg__SvgStyle-pyey0i-0 gYARyh"
    fill="none"
    height="24px"
    size={null}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="18"
      x2="6"
      y1="6"
      y2="18"
    />
    <line
      x1="6"
      x2="18"
      y1="6"
      y2="18"
    />
  </svg>
</button>
`;
